app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetMedicationFragment.java

```diff
@@ -0,0 +1,391 @@
+        });
+    }
+
+    /**
+     * Method responsible for initializing the string for the inputMedicationDate button.
+     * @param medicationIniDate The date of the medication
+     */
+    private void showMedicationIniDate(DateTime medicationIniDate) {
+        StringBuilder dateString = new StringBuilder();
+        dateString.append(medicationIniDate.getYear()).append(DATESEPARATOR);
+        if (medicationIniDate.getMonth() < FIRST_TWO_DIGITS) {
+            dateString.append('0');
+        }
+        dateString.append(medicationIniDate.getMonth()).append(DATESEPARATOR);
+        if (medicationIniDate.getDay() < FIRST_TWO_DIGITS) {
+            dateString.append('0');
+        }
+        dateString.append(medicationIniDate.getDay());
+        this.medicationDate.setText(dateString);
+    }
+
+    /**
+     * Initialize the edit medication button.
+     */
+    private void initializeEditMedicationButton() {
+        editMedicationButton.setOnClickListener(v -> {
+            if (isAnyFieldBlank()) {
+                showErrorMessage();
+            } else {
+                if (editing) {
+                    initializeEditButtonListener();
+                } else {
+                    initializeAddButtonListener();
+                }
+
+                dialog.dismiss();
+                initializeMedicationsLayoutView();
+            }
+        });
+    }
+
+    /**
+     * Show the error message.
+     */
+    private void showErrorMessage() {
+        Toast errorMsg = Toast.makeText(getActivity(), R.string.empty_fields, Toast.LENGTH_LONG);
+        errorMsg.show();
+    }
+
+    /**
+     * Method responsible for initializing the addButton listener.
+     */
+    private void initializeAddButtonListener() {
+        DateTime medicationIniDate = getDateTime();
+        String medicationName = Objects.requireNonNull(inputMedicationName.getText()).toString();
+        double medicationQuantity = Double.parseDouble(Objects.requireNonNull(
+            inputMedicationQuantity.getText()).toString());
+        int medicationPeriodicity = Integer.parseInt(Objects.requireNonNull(
+            inputMedicationPeriodicity.getText()).toString());
+        int medicationDuration = Integer.parseInt(Objects.requireNonNull(
+            inputMedicationDuration.getText()).toString());
+        medication = new Medication(medicationName, medicationQuantity, medicationPeriodicity,
+            medicationDuration, medicationIniDate);
+        communicateAddition();
+    }
+
+    /**
+     * Method responsible for communicating the pet addition to the server.
+     */
+    private void communicateAddition() {
+        try {
+            InfoPetFragment.getCommunication().addPetMedication(pet, medication);
+        } catch (MedicationAlreadyExistingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Method responsible for initializing the editButton listener.
+     */
+    private void initializeEditButtonListener() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method length is 20 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5421176)